src/loaders/TextureLoader.js

```diff
@@ -17,6 +17,7 @@ class TextureLoader extends Loader {
 		const loader = new ImageLoader( this.manager );
 		loader.setCrossOrigin( this.crossOrigin );
 		loader.setPath( this.path );
+		loader.setWithCredentials( this.withCredentials );
```

Comment:
`withCredentials` is not used by `ImageLoader` so how does this line has any effect?

`withCredentials` is only relevant for `FileLoader`.

Reply:
Hi @Mugen87 , I'm not an expert of three.js so I cannot really answer your question, I just faced such problem on google/model-viewer repo and by debugging on console I saw in that point `this.withCredentials` was correctly set but it wans't on the new instance of `ImageLoader` created on load method. Could it be the problem is model-viewer is using a wrong loader? Should it use a file loader instead of a texture loader?

cc @elalish

Reply:
This is where the loader is created: 

https://github.com/google/model-viewer/blob/84f3b190445031f2323146bafe7c3523ea3114d7/packages/model-viewer/src/three-components/TextureUtils.ts#L66

Reply:
Sorry, I can't say something about model-viewer. The only thing I can say for sure is that this PR has no effect in the `three.js` codebase.

Reply:
![Peek 16-02-2023 09-53](https://user-images.githubusercontent.com/11502495/219315870-4fff3d9a-39fc-47df-9dee-828320eb5b0d.gif)

Reply:
Basically the error here seems to be that I didn't provided an hdr image, If that image would be hdr the hdrLoader is used and so `withCredentials` would be set correctly.

Question is (for @elalish ) when skybox-image isn't an hdr image, what loader should we use that also provides `withCredentials` support as ldrLoader now uses TextureLoader that doesn't have such feature?

Question for @Mugen87 is there any reason why you don't provide `withCredentials` support to the TextureLoader?

Reply:
> Question for @Mugen87 is there any reason why you don't provide withCredentials support to the TextureLoader?

Yes. `ImageLoader` does not user `FileLoader` for a serious of reasons. They are mentioned in #10439.

`ImageLoader` uses the `Image` HTML element instead which has not credentials support though.

Reply:
OK so a possible (simple) solution on model-viewer could be to force use hdrLoader when `withCredentials` is set to true even if the image is not in hdr, in this way `withCredentials` will be correctly set in the request.

Need to speak with @elalish to see if this could have any side effect I don't know but should work

Thanks @Mugen87, closing this

Reply:
> OK so a possible (simple) solution on model-viewer could be to force use hdrLoader when withCredentials is set to true even if the image is not in hdr, in this way withCredentials will be correctly set in the request.

I don't think that works since `RGBELoader` expects HDR images. It can't handle typical image formats like JPG, PNG etc..

Reply:
Other loader that support ldr images and also withCredentials?

Reply:
What type of images do you want to load?

Reply:
I think `ImageBitmapLoader` could support `withCredentials` but there are some consequences of a custom `TextureLoader` that internally uses `ImageBitmapLoader`. For example texture properties like `flipY` do not work with image bitmaps.